src/Core/IdentityServer/ClientStore.cs

```diff
@@ -106,6 +110,42 @@ public async Task<Client> FindClientByIdAsync(string clientId)
                     }
                 }
             }
+            else if (clientId.Contains("user."))
+            {
+                var idParts = clientId.Split('.');
+                if (idParts.Length > 2 && Guid.TryParse(idParts[2], out var id))
+                {
+                    var user = await _userRepository.GetByIdAsync(id);
+                    if (user != null)
+                    {
+                        var cliUris = new List<string>();
+                        for (var port = 8065; port <= 8070; port++)
+                        {
+                            cliUris.Add(string.Format("http://localhost:{0}", port));
+                        }
+
+                        return new Client
+                        {
+                            ClientId = clientId,
+                            RequireClientSecret = true,
+                            ClientSecrets = { new Secret(user.ApiKey.Sha256()) },
+                            AllowedScopes = new string[] { "api" },
+                            AllowedGrantTypes = GrantTypes.ClientCredentials,
+                            AccessTokenLifetime = 3600 * 1,
+                            RedirectUris = cliUris,
+                            PostLogoutRedirectUris = cliUris,
+                            Claims = new List<ClientClaim>
+                            {
+                                new ClientClaim(JwtClaimTypes.Subject, user.Id.ToString()),
+                                new ClientClaim(JwtClaimTypes.Email, user.Email),
+                                new ClientClaim(JwtClaimTypes.EmailVerified, user.EmailVerified.ToString()),
+                                new ClientClaim(JwtClaimTypes.Name, user.Name),
+                                new ClientClaim(JwtClaimTypes.AuthenticationMethod, "Application", "external")
```

Comment:
Not sure I understand why these are needed. Shouldn't the profile service handle this?

Reply:
I'm not sure at what point the ProfileService stuff gets called (maybe I'm missing an implementation that needs to be added for the new auth flow?), but right now it doesn't seem to be adding any of these claims. Without adding the claims here the token I get back is empty.

Reply:
I am concerned that not all claims are being added. There are many more than this such as orgs, premium, etc. Will need to troubleshoot further.

Reply:
It looks like the ProfileService logic only runs when a direct `User` is involved, meaning it won't run when using a `client_credentials` grant type. I'm not sure there's a way to get around that. 

Would an acceptable solution be adding the claims here that are missing from the profile service? Or moving that logic to a BW service we can control, and referencing that in the ProfileService and in this Client initialization?

Reply:
I went with adding them directly to this client for now, since there were some differences between what this should look like and what the ProfileService logic is doing, but all claims added in ProfileService should be accounted for now.